spec/presenters/users/devices/devices_show_presenter_spec.rb

```diff
@@ -0,0 +1,109 @@
+require "rails_helper"
+
+describe ::Users::Devices::DevicesShowPresenter do
+  subject(:show_presenter) { described_class.new(user, id: device.id) }
+  let(:user) { FactoryGirl.create(:user) }
+  let(:device) { FactoryGirl.create(:device, user_id: user.id) }
+  let(:checkins) do
+    FactoryGirl.create(:checkin, device_id: device.id)
+    FactoryGirl.create(:checkin, device_id: device.id).reverse_geocode!
+    FactoryGirl.create(:checkin, device_id: device.id)
+  end
+
+  describe "Interface" do
+    %i(user device checkins filename show_gon form_for form_path form_range_filter).each do |method|
+      it { is_expected.to respond_to method }
+    end
+  end
+
+  describe "user" do
+    it "returns user" do
+      expect(show_presenter.user).to eq user
+    end
+  end
+
+  describe "device" do
+    it "returns a device" do
+      expect(show_presenter.device).to be_kind_of Device
+    end
+  end
+
+  describe "checkins" do
+    it "returns nil if no download params" do
+    end
+
+    it "returns checkins converted to param provided" do
+      checkins
```

Comment:
Even though I only need 'checkins' in two tests? seems like overkill to use let! and have them loaded for every test. Or do you mean put the let! inside the context where the checkins are needed?